src/bootstrap.js

```diff
@@ -0,0 +1,133 @@
+import * as algolia from './algolia/index.js';
+import config from './config.js';
+import datadog from './datadog.js';
+import log from './log.js';
+import ms from 'ms';
+import * as npm from './npm/index.js';
+import saveDocs from './saveDocs.js';
+
+let loopStart;
+
+/**
+ * Bootstrap is the mode that goes from 0 to all the packages in NPM
+ * In other word it is reindexing everything from scratch.
+ *
+ * It is useful if:
+ *  - you are starting this project for the first time
+ *  - you messed up with your Algolia index
+ *  - you lagged too much behind
+ *
+ * Watch mode should/can be reliably left running for weeks/months as CouchDB is made for that.
```

Comment:
mmh was not thinking about that true.
maybe instead of relaunching bootstrap we should have a parallel process that does this update, because during the bootstrap we are not able to process new changes.

Reply:
Yes, that's a pretty interesting strategy, and would avoid the memory peaks we have during bootstrap, since it would become very rare. Let's add a comment on the reason for redoing bootstrap for now, and do our parallel processing in a separate iteration!